File: The Office/VisualBasic/TheOffice/Worker.vb
Public Class Worker

    Public Property Name As String
    Public Property Room As Room

    Public Sub New(Name As String, Room As Room)
        Me.Name = Name
        Me.Room = Room
    End Sub

    Public Sub Move()
        Dim destination As Room = Me.Room.GetRandomExit()
        Me.Room.Workers.Remove(Me)
        destination.Workers.Add(Me)
        Me.Room = destination
    End Sub

End Class

File: The Office/CSharp/TheOffice/Worker.cs
namespace TheOffice
{
    public class Worker
    {
        public string Name { get; set; }
        public Room Room { get; set; }

        public Worker(string name, Room room)
        {
            this.Name = name;
            this.Room = room;
        }

        public void Move()
        {
            Room destination = this.Room.GetRandomExit();
            this.Room.Workers.Remove(this);
            destination.Workers.Add(this);
            this.Room = destination;
        }
    }
}